src/output_panel.c

```diff
@@ -632,26 +639,31 @@ static gboolean paperstrip_draw_event(GtkWidget *widget, cairo_t *c, struct outp
 	cairo_set_source(c,stopped?yellow:white);
 	for(i = snst->events_wp;;) {
 		if(!snst->events_count || snst->events[i]==NULL_EVENT_TIME) break;
-		double event = now - absEventTime(snst->events[i]) + snst->trace_centering + sweep * PAPERSTRIP_MARGIN / (2 * zoom_factor);
+		double event = now - absEventTime(snst->events[i]) + snst->trace_centering  + sweep * PAPERSTRIP_MARGIN / (2 * zoom_factor);
 		if(!stopped){
 			cairo_set_source(c, event_is_TIC_or_TOC(snst->events[i])==TIC?ticColor:tocColor);
 		}
-		int column = floor(fmod(event, (sweep / zoom_factor)) * strip_width / (sweep / zoom_factor));
-		int row = floor(event / sweep);
+		double phase = fmod(event, sweep) / sweep;   // 0.0 -> 1.0
+		double cycle =  strip_width * ( 0.5 + (phase - 0.5) * zoom_factor);
+		int column = floor(fmod(cycle, strip_width));
```

Comment:
This zoom only works if `sweep` is a integer multiple of `strip_width`.  If not, what happens is `phase = fmod(event, sweep)` will increase/decrease until it reaches zero and wraps around.  At the same time, `column = fmod(cycle, strip_width)` is also wrapping around as the path cross the chart edges.  If `column` doesn't wrap around at the same time as `phase`, then the path will jump from some point in the middle of the chart, by an amount equal to fmod(sweep, strip_width)`.  This happens if sweep isn't an exact multiple of strip_width.

In my PR that adds a smooth beat length zoom I wrote a totally new algorithm that solves this problem.